pkg/provider/podsTracker.go

```diff
@@ -150,7 +150,8 @@ func (pt *PodsTracker) processPodUpdates(ctx context.Context, pod *v1.Pod) bool
 			pod.Status.Message = statusMessageNotFound
 			now := metav1.NewTime(time.Now())
 			for i := range pod.Status.ContainerStatuses {
-				if pod.Status.ContainerStatuses[i].State.Running == nil {
+				if pod.Status.ContainerStatuses != nil &&
```

Comment:
Why do you need the nil check for ContainerStatuses? If it is nil, the range based for loop will be no-op. The program cannot reach here.

Reply:
removed